runtime/oti/j9protos.h

```diff
@@ -1133,6 +1133,52 @@ extern J9_CFUNC struct J9Method* jitResolveStaticMethodRef(J9VMThread *vmStruct,
 extern J9_CFUNC struct J9Method* jitResolveSpecialMethodRef(J9VMThread *vmStruct, J9ConstantPool *constantPool, UDATA cpOrSplitIndex, UDATA resolveFlags);
 extern J9_CFUNC struct J9Class * jitGetDeclaringClassOfROMField(J9VMThread *vmStruct, J9Class *clazz, J9ROMFieldShape *romField);
 
+typedef struct J9MethodFromSignatureWalkState {
+	U_8 *className;
+	U_16 classNameLength;
+	struct J9JNINameAndSignature nameAndSig;
+	struct J9VMThread *vmThread;
+	struct J9ClassLoaderWalkState classLoaderWalkState;
+} J9MethodFromSignatureWalkState;
+
+/**
+ * Cleans up the iterator after matching all methods found.
+ *
+ * @param state				The inlialized iterator state
+ */
+extern J9_CFUNC void
+allMethodFromSignatureEndDo(J9MethodFromSignatureWalkState *state);
+
+/**
+ * Advances the iterator looking for the next method matching the full signature provided during initialization.
+ *
+ * @param state				The inlialized iterator state
+ *
+ * @return 					J9Method matching the full signature provided during initialization of the iterator in the
+ * 							class loader if it exists; NULL if no such method is found.
+ */
+extern J9_CFUNC struct J9Method*
+allMethodFromSignatureNextDo(J9MethodFromSignatureWalkState *state);
+
+/**
+ * Initializes the iterator state and returns the first method matching the supplied signature from any class loader.
+ *
+ * @param state				The iterator state which will be initialized.
+ * @param vm				The VM instance to look for methods in.
+ * @param flags				The flags forwarded to class loader iterator functions
+ * @param className			The class name including the package.
+ * @param classNameLength	The length (in number of bytes) of the class name
+ * @param methodName		The method name
+ * @param methodNameLength	The length (in number of bytes) of the method name
+ * @param methodSig			The method signature
+ * @param methodSigLength	The length (in number of bytes) of the method signature
+ *
+ * @return 					J9Method matching the full signature provided in the first class loader it is found; NULL if
+ * 							no such method is found.
+ */
+extern J9_CFUNC struct J9Method*
+allMethodFromSignatureStartDo(J9MethodFromSignatureWalkState *state, J9JavaVM* vm, UDATA flags, U_8 *className, U_16 classNameLength, U_8 *methodName, U_16 methodNameLength, U_8 *methodSig, U_16 methodSigLength);
```

Comment:
The types of `className`, `methodName` and `methodSig` should all be `const U8 *`.

Reply:
See above.